src/main/java/com/artipie/Settings.java

```diff
@@ -74,6 +76,13 @@
      */
     CompletionStage<Credentials> credentials();
 
+    /**
+     * Artipie global permissions.
+     *
+     * @return Global permissions if configured, empty - otherwise.
+     */
+    CompletionStage<Optional<GlobalPermissions>> permissions();
```

Comment:
@olegmoz it seems to me that using `Optional` as a return type always adds some unnecessary complications to the code. As far as I understand the idea, absence of global permissions means that we simply use repo permissions as we do now. May be for this case we can simply create and use some special implementation (which simply returns permissions for repository) of `GlobalPermissions` and avoid returning `Optional` here?

Reply:
@olenagerasimova I agree that it adds additional complexity, but we need special handling of `GlobalPermissions` absence. See the table in comments of https://github.com/artipie/artipie/pull/549